psycopg/utils.c

```diff
@@ -124,6 +124,50 @@ psycopg_escape_identifier_easy(const char *from, Py_ssize_t len)
     return rv;
 }
 
+char *
+psycopg_escape_conninfo(const char *from, Py_ssize_t len)
+{
+    char *rv = NULL;
+    const char *src;
+    const char *end;
+    char *dst;
+    int space = 0;
+
+    if (!len) { len = strlen(from); }
+    end = from + len;
+
+    if (!(rv = PyMem_Malloc(3 + 2 * len))) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+
+    /* check for any whitespace or empty string */
+    if (from < end && *from) {
+        for (src = from; src < end && *src; ++src) {
```

Comment:
By the way, there seems to be a missing check on `len` in `psycopg_escape_identifier_easy()` just above this function:  if `len < strlen(from)` it will try to go beyond provided `len`.